panther_battery/src/battery_publisher.cpp

```diff
@@ -0,0 +1,78 @@
+#include <panther_battery/battery_publisher.hpp>
+
+#include <memory>
+#include <stdexcept>
+
+#include <rclcpp/rclcpp.hpp>
+
+namespace panther_battery
+{
+
+BatteryPublisher::BatteryPublisher(std::shared_ptr<rclcpp::Node> node) : node_(node)
```

Comment:
There can be multiple instances of BatteryPublisher (in case of 2 batteries) so I don't think std::move should be used here. It should point to rcpcpp::Node in main node.